realm/realm-library/src/main/java/io/realm/RealmFieldType.java

```diff
@@ -37,7 +37,8 @@
     BINARY(4),
     UNSUPPORTED_TABLE(5),
     UNSUPPORTED_MIXED(6),
-    DATE(7),
+    UNSUPPORTED_DATE(7),
```

Comment:
As noted above, this enum is kept in sync with a similar enum in core. I rather not removed the three unused/unsupported types.